src/main/resources/static/js/shopBookstore.js

```diff
@@ -0,0 +1,135 @@
+                },
+                success: function (data) {
+                    // populate table
+                    populateBooksTable(data);
+                },
+                error: function (err) {
+                    alert("Failed to get books with err:" + err);
+                }
+            })
+        }
+
+        var populateBooksTable = function (books) {
+            let booksTable = $('#books');
+            let booksMessage = $("#booksMessage");
+            booksTable.empty();
+    
+            if (books == null || books.length == 0) {
+                $(booksMessage).show();
+                $(booksTable).hide();
+                return;
+            }
+            $(booksMessage).hide();
+    
+            booksTable.append('<tr> ' +
+                '<th>ID</th> ' +
+                '<th>Name</th>' +
+                '<th>ISBN</th>' +
+                '<th>Picture</th>' +
+                '<th>Description</th>' +
+                '<th>Author</th>' +
+                '<th>Publisher</th>' +
+                '<th>Available</th>' +
+                '<th></th>' +
+                '</tr>');
+    
+            $.each(books, function (indx, book) {
+                let buttonId = 'delete-' + indx;
+                if(book.available){
+                booksTable.append('<tr>' +
+                    '<td>' + book.id + '</td>' +
+                    '<td>' + book.name + '</td>' +
+                    '<td>' + book.isbn + '</td>' +
+                    '<td>' + book.picture + '</td>' +
+                    '<td>' + book.description + '</td>' +
+                    '<td>' + book.author + '</td>' +
+                    '<td>' + book.publisher + '</td>' +
+                    '<td>' + book.available + '</td>' +
+                    '<td><button id="' + buttonId + '" class="button3">Add to Cart</button></td>' +
+                    '</tr>');
+                $('#' + buttonId).click(function () {
+                    addBookToCart(book);
+                });
+            }
+            });
+            $(booksTable).show();
+        };
+
+        var addBookToCart = function(bookToAdd){
+            var call = $.ajax({
+                type: "PUT",
+                url: "/api/customers/"+urlParam('customerId')+'/shoppingcart',
+                data: JSON.stringify(bookToAdd), // serializes the form's elements.
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                success: function (data) {
+                    window.location.href='/viewCustomer?customerId='+urlParam('customerId')
+                    
+                }
+                ,
+                error: function (err) {
+                    alert("Book is already in cart");
+                }
+            });
+        
+        }
+
+        populateBooks();
+        populateBookstoreInfo();
+        populateCustomerInfo();
+    })
```

Comment:
Javascript doesn't like not having a newline at the end of .js files, don't think theres any functional penalties though